gno.land/pkg/gnoclient/client_test.go

```diff
@@ -1,100 +1,377 @@
+				&mockSigner{},
+				nil,
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      100000,
+				GasFee:         "10000ugnot",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
+			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "random/path",
+					FuncName: "RandomName",
+					Send:     "",
+					Args:     []string{},
+				},
+			},
+			expectedError: ErrMissingRPCClient,
+		},
+		{
+			name: "Invalid Gas Fee",
+			client: Client{
+				Signer:    &mockSigner{},
+				RPCClient: &mockRPCClient{},
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      100000,
+				GasFee:         "",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
 			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "random/path",
+					FuncName: "RandomName",
+				},
+			},
+			expectedError: ErrInvalidGasFee,
+		},
+		{
+			name: "Negative Gas Wanted",
+			client: Client{
+				Signer:    &mockSigner{},
+				RPCClient: &mockRPCClient{},
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      -1,
+				GasFee:         "10000ugnot",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
+			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "random/path",
+					FuncName: "RandomName",
+					Send:     "",
+					Args:     []string{},
+				},
+			},
+			expectedError: ErrInvalidGasWanted,
+		},
+		{
+			name: "0 Gas Wanted",
+			client: Client{
+				Signer:    &mockSigner{},
+				RPCClient: &mockRPCClient{},
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      0,
+				GasFee:         "10000ugnot",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
+			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "random/path",
+					FuncName: "RandomName",
+					Send:     "",
+					Args:     []string{},
+				},
+			},
+			expectedError: ErrInvalidGasWanted,
 		},
+		{
+			name: "Invalid PkgPath",
+			client: Client{
+				Signer:    &mockSigner{},
+				RPCClient: &mockRPCClient{},
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      100000,
+				GasFee:         "10000ugnot",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
+			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "",
+					FuncName: "RandomName",
+					Send:     "",
+					Args:     []string{},
+				},
+			},
+			expectedError: ErrEmptyPkgPath,
+		},
+		{
+			name: "Invalid FuncName",
+			client: Client{
+				Signer:    &mockSigner{},
+				RPCClient: &mockRPCClient{},
+			},
+			cfg: BaseTxCfg{
+				GasWanted:      100000,
+				GasFee:         "10000ugnot",
+				AccountNumber:  1,
+				SequenceNumber: 1,
+				Memo:           "Test memo",
+			},
+			msgs: []MsgCall{
+				{
+					PkgPath:  "random/path",
+					FuncName: "",
+					Send:     "",
+					Args:     []string{},
+				},
+			},
+			expectedError: ErrEmptyFuncName,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			res, err := tc.client.Call(tc.cfg, tc.msgs...)
+			assert.Equal(t, err, tc.expectedError)
```

Comment:
You should use `assert.ErrorIs` here

Reply:
[a4bf5cb](https://github.com/gnolang/gno/pull/1565/commits/a4bf5cbe41ee55a51d8e268d81a70f1e3f42e173)

Reply:
It's still not `assert.ErrorIs`, but just `errors.Is`

Reply:
My bad: [f6e78d2](https://github.com/gnolang/gno/pull/1565/commits/f6e78d2e07aac815e9bb71d0a6a9069086f25e7f)